pkg/recovery/etcd.go

```diff
@@ -153,28 +152,22 @@ func (s *etcdSelfhostedBackend) read(ctx context.Context) (*controlPlane, error)
 		return nil, err
 	}
 
-	d, err := s.getBytes(ctx, etcdTPRKey)
+	d, err := s.getBytes(ctx, etcdCRDKey)
 	if err != nil {
 		return nil, err
 	}
 
-	var tpr v1beta1.ThirdPartyResourceData
-	err = decode(s.decoder, d, &tpr)
+	var kubeetcd spec.EtcdCluster
+	err = json.Unmarshal(d, &kubeetcd)
```

Comment:
So something like:
```
err = json.Unmarshal(d, &kubeetcd)
if err != nil
    err = proto.Unmarshal(d, &kubeetcd)
    if != nil {
        return nil, err
    }
```
How can I know if it is profobuf? I didn't knew that k8s could use protobuf.